docs/tutorial/overview.md

```diff
@@ -28,33 +27,112 @@ component main() -> {
 
 ### `Decl` nodes
 
-`Decl` nodes instantiate new memories and registers. Here's FuTIL code that creates a new memory `foo`:
+`Decl` nodes instantiate new memories and registers. Here's FuTIL code that creates a new memory `foo`, with 4 32-bit elements and a 32-bit indexor:
+
 ```
 foo = prim std_mem_d1(32, 4, 32);
 ```
 
-For each `Decl` node, we just determine if we're instantiating a memory or a register, and then translate that to a corresponding FuTIL declaration and place that inside the `cells` section of our generated program.
+For each `Decl` node, we need to determine if we're instantiating a memory or a register, and then translate that to a corresponding FuTIL declaration and place that inside the `cells` section of our generated program. To get you started, here's some code that walks through each register declaration and generates a FuTIL program with those registers:
+
+```
+def emit_reg_decl(name, size):
+    return "{} = prim std_reg({});".format(name, 32)
+
+# Collect memory and register declarations.
+cells = []
+for decl in prog.decls:
+    if decl.type.size:  # A memory
+        ...
+    else:  # A register
+        cells.append(emit_reg_decl(decl.name, 32))
+
+futil_program = '''
+component main() -> {{
+  cells {{ {} }}
+  wires {{ }}
+  control {{ }}
+}}
+'''.format("\n".join(cells))
+```
 
 ### `Map` and `Reduce` nodes
 
 For every map or reduce node, we need to generate FuTIL code that iterates over an array, performs some kind of computation, and then stores the result of that computation in another array if we're doing a map, or in an accumulator register if we're doing a reduce. We'll need to generate a few FuTIL groups:
 - `incr_idx`: Increments an `idx` register using an adder. This group is done when the `idx` register is written to.
-- `cond`: Applies a "less than" operator to `idx`, and the length of our input arrays.
+- `cond`: Applies a "less than" operator to `idx`, and the length of our input arrays, using the `le` hardware unit.
 - `eval_body`: Reads from an array, performs some kind of computation, and writes the result of the computation to an accumulator register or another array.
 
-We'll make these groups for each `Map` and `Reduce` node, so to avoid naming collisions, we'll suffix each group with an integer starting at 0, incrementing each time we need to add new sets of  groups. These groups will be added to the `wires` section. We'll also need to add logic to the `control` section as well that uses these groups to process arrays:
+We'll make these groups for each `Map` and `Reduce` node, so to avoid naming collisions, we'll suffix each group with an integer starting at 0, incrementing each time we need to add a new set of  groups. These groups will be added to the `wires` section. We'll also need to add logic to the `control` section as well that uses these groups to process arrays:
+
 ```
 while le0.out with cond0 {
   seq { eval_body0; incr_idx0; }
 }
 ```
 
+### Parallelization
```

Comment:
Before jumping into this, maybe refer to a full example and show people how to run it using `fud.